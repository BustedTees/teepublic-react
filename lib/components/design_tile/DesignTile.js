var _jsxFileName = 'src/lib/components/design_tile/DesignTile.js';

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import Row from '../row/Row';
import Button from '../button/Button';
import Column from '../column/Column';
import MoneyHelper from '../../utils/MoneyHelper';

import './DesignTile.css';

var CLASS_ROOT = 'tp-design-tile';

var DESIGN_IMAGE = {
  small: 200,
  medium: 300,
  large: 400
};

var AVATAR_IMAGE = {
  small: 26,
  medium: 40,
  large: 53
};

var DesignTile = function (_Component) {
  _inherits(DesignTile, _Component);

  function DesignTile() {
    _classCallCheck(this, DesignTile);

    return _possibleConstructorReturn(this, (DesignTile.__proto__ || Object.getPrototypeOf(DesignTile)).apply(this, arguments));
  }

  _createClass(DesignTile, [{
    key: 'randomCanvas',
    value: function randomCanvas(canvases) {
      var rand_canvas = Math.floor(Math.random() * (canvases.length - 1)) + 1;
      return canvases[rand_canvas];
    }
  }, {
    key: 'findCanvas',
    value: function findCanvas(canvases, canvas_slug) {
      canvases.forEach(function (canvas) {
        if (canvas.slug == canvas_slug) {
          return canvas;
        }
      });
      return this.randomCanvas(canvases);
    }
  }, {
    key: 'render',
    value: function render() {
      var _props = this.props,
          className = _props.className,
          design = _props.design,
          designer = _props.designer,
          size = _props.size,
          props = _objectWithoutProperties(_props, ['className', 'design', 'designer', 'size']);

      var classes = classnames(CLASS_ROOT, _defineProperty({}, CLASS_ROOT + '--' + size, size), className);

      var canvas = this.randomCanvas(design.canvases);
      var product_url = 'https://www.teepublic.com/' + canvas.slug + '/' + design.slug;

      // const designImage = <img className={`${CLASS_ROOT}__image`} src={design.image_url} alt="Design Image" height={DESIGN_IMAGE[size]}/>
      var designImage = React.createElement('img', {
        className: CLASS_ROOT + '__image',
        src: canvas.mockup_url,
        alt: 'Design Image',
        height: DESIGN_IMAGE[size],
        __source: {
          fileName: _jsxFileName,
          lineNumber: 59
        },
        __self: this
      });
      var designTitle = React.createElement(
        'a',
        { className: CLASS_ROOT + '__title', href: product_url, target: '_blank', __source: {
            fileName: _jsxFileName,
            lineNumber: 67
          },
          __self: this
        },
        design.title
      );
      var designPrice = React.createElement(
        'p',
        { className: CLASS_ROOT + '__price', __source: {
            fileName: _jsxFileName,
            lineNumber: 72
          },
          __self: this
        },
        new MoneyHelper(canvas.prices.regular_price, 'USD').commaSeprated()
      );
      var designerLocation = React.createElement(
        'p',
        { className: CLASS_ROOT + '__location', __source: {
            fileName: _jsxFileName,
            lineNumber: 77
          },
          __self: this
        },
        designer.location
      );
      var designerAvatar = React.createElement(
        'a',
        { href: designer.store_url, target: '_blank', __source: {
            fileName: _jsxFileName,
            lineNumber: 80
          },
          __self: this
        },
        React.createElement('img', {
          className: CLASS_ROOT + '__avatar',
          src: designer.avatar_url,
          alt: 'Designer Avatar',
          height: AVATAR_IMAGE[size],
          __source: {
            fileName: _jsxFileName,
            lineNumber: 81
          },
          __self: this
        })
      );
      var buyButton = React.createElement(
        Button,
        { style: 'fill', size: 'small', __source: {
            fileName: _jsxFileName,
            lineNumber: 90
          },
          __self: this
        },
        'Buy'
      );

      return React.createElement(
        'a',
        { className: classes, href: product_url, target: '_blank', __source: {
            fileName: _jsxFileName,
            lineNumber: 96
          },
          __self: this
        },
        React.createElement(
          Column,
          { justify: 'start', align: 'center', __source: {
              fileName: _jsxFileName,
              lineNumber: 97
            },
            __self: this
          },
          designImage,
          React.createElement(
            Row,
            {
              className: CLASS_ROOT + '__details',
              justify: 'between',
              align: 'start',
              __source: {
                fileName: _jsxFileName,
                lineNumber: 99
              },
              __self: this
            },
            React.createElement(
              Column,
              { justify: 'start', align: 'start', __source: {
                  fileName: _jsxFileName,
                  lineNumber: 104
                },
                __self: this
              },
              React.createElement(
                Row,
                { justify: 'start', align: 'center', __source: {
                    fileName: _jsxFileName,
                    lineNumber: 105
                  },
                  __self: this
                },
                designerAvatar,
                React.createElement(
                  Column,
                  {
                    className: CLASS_ROOT + '__title-location',
                    justify: 'start',
                    align: 'start',
                    __source: {
                      fileName: _jsxFileName,
                      lineNumber: 107
                    },
                    __self: this
                  },
                  designTitle,
                  designerLocation
                )
              )
            ),
            React.createElement(
              Column,
              { justify: 'start', align: 'end', __source: {
                  fileName: _jsxFileName,
                  lineNumber: 117
                },
                __self: this
              },
              designPrice,
              buyButton
            )
          )
        )
      );
    }
  }]);

  return DesignTile;
}(Component);

export default DesignTile;


DesignTile.propTypes = {
  size: PropTypes.oneOf(['small', 'medium', 'large'])
};

DesignTile.defaultProps = {
  size: 'large'
};