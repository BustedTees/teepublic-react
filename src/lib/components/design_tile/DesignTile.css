.tp-design-tile {
  text-decoration: none;
}
.tp-design-tile__location {
  color: #616161;
  max-width: 100%;
  text-overflow: ellipsis;
  overflow: hidden;
}

.tp-design-tile__title {
  color: #19b1ff;
  max-width: 100%;
  text-overflow: ellipsis;
  overflow: hidden;
}

.tp-design-tile__title-location {
  white-space: nowrap;
}

.tp-design-tile__price {
  margin: 0;
  color: gray;
}

.tp-design-tile--small {
  margin: 5px;
}
.tp-design-tile--small .tp-design-tile__title-location {
  width: 100px;
}
.tp-design-tile--small .tp-design-tile__details {
  width: 200px;
  margin-top: 5px;
}
.tp-design-tile--small .tp-design-tile__avatar {
  border-radius: 50%;
  margin: 0 5px;
}
.tp-design-tile--small .tp-design-tile__location {
  font-size: 12px;
  margin: 2px 0;
}
.tp-design-tile--small .tp-design-tile__price {
  font-size: 16px;
}

.tp-design-tile--medium {
  margin: 10px;
}
.tp-design-tile--medium .tp-design-tile__title-location {
  width: 170px;
}
.tp-design-tile--medium .tp-design-tile__details {
  width: 300px;
  margin-top: 7px;
}
.tp-design-tile--medium .tp-design-tile__avatar {
  border-radius: 50%;
  margin: 0 5px;
}
.tp-design-tile--medium .tp-design-tile__location {
  font-size: 12px;
  margin: 3px 0;
}
.tp-design-tile--medium .tp-design-tile__price {
  font-size: 20px;
}

.tp-design-tile--large {
  margin: 15px;
}
.tp-design-tile--large .tp-design-tile__title-location {
  width: 250px;
}
.tp-design-tile--large .tp-design-tile__details {
  width: 400px;
  margin-top: 10px;
}
.tp-design-tile--large .tp-design-tile__avatar {
  border-radius: 50%;
  margin: 0 7px;
}
.tp-design-tile--large .tp-design-tile__location {
  font-size: 15px;
  margin: 4px 0;
}
.tp-design-tile--large .tp-design-tile__price {
  font-size: 25px;
}
