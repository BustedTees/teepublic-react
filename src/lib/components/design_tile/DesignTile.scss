@import '../variables';

.tp-design-tile {
  text-decoration: none;
  margin-bottom: 20px;
}

.tp-design-tile__buy-btn {
  width: 100%;
  border-radius: 2px;
  background: $color-blue;
  height: 35px;
  color: white;
  font-size: 14px;
  font-weight: bold;
  width: 56px;
  display: block;
  border: none;
  border-bottom: 2px solid $color-blue-dark;
  padding-top: 5px;
  margin-left: 20px;
  cursor: pointer;
}

.tp-design-tile__details {
  width: 100%;
  display: flex;
  justify-content: space-between;
  margin-top: 10px;
  margin-bottom: 10px;
}

.tp-design-tile__image {
  display: block;
}

.tp-design-tile__owner {
  color: #616161;
  max-width: 100%;
  text-overflow: ellipsis;
  overflow: hidden;
  margin: 2px 0;
}

.tp-design-tile__price {
  margin: 0;
  font-size: 24px;
  font-weight: bold;
}

.tp-design-tile__title {
  color: #19b1ff;
  max-width: 100%;
  text-overflow: ellipsis;
  overflow: hidden;
  margin: 0;
}

.tp-design-tile__title-owner {
  white-space: nowrap;
}
