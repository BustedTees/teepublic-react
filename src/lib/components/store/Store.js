import React, { Component } from 'react';

import PropTypes from 'prop-types';
import classnames from 'classnames';

import DesignCollection from '../design_collection/DesignCollection';
import CartButton from '../cart_button/CartButton';
import Pagination from '../pagination/Pagination';
import StoreFilter from '../store_filter/StoreFilter';

import './Store.scss';

const CLASS_ROOT = 'tp-store';

export default class Store extends Component {
  constructor(props) {
    super(props);

    this.state = {
      selectedPage: props.selectedPage || 1,
      selectedAlbumId: parseInt(props.selectedAlbumId, 10) || null,
      selectedProductTypeName: props.selectedProductTypeName
    };
  }

  pageChangeHandler = page => {
    this.setState({ selectedPage: page });
    const { selectedAlbumId, selectedProductTypeName } = this.state;
    this.fetchStore(page, selectedAlbumId, selectedProductTypeName);
  };

  albumChangeHandler = albumId => {
    this.setState({ selectedAlbumId: albumId });
    const { selectedPage, selectedProductTypeName } = this.state;
    this.fetchStore(selectedPage, albumId, selectedProductTypeName);
  };

  productTypeChangeHandler = productTypeName => {
    this.setState({ selectedProductTypeName: productTypeName });
    const { selectedPage, selectedAlbumId } = this.state;
    this.fetchStore(selectedPage, selectedAlbumId, productTypeName);
  };

  fetchStore = (pageNum, albumId, productTypeName) => {
    const { id } = this.props.storeData;

    window.location = this.props.configuration.storeUrl(
      id,
      pageNum,
      albumId,
      productTypeName
    );
  };

  render() {
    const { designs, albums, productTypes } = this.props.storeData._embedded;
    const {
      selectedPage,
      selectedAlbumId,
      selectedProductTypeName
    } = this.state;
    const totalPages = parseInt(this.props.storeData.totalPages, 10);

    const { className } = this.props;
    const classes = classnames(className, CLASS_ROOT, 'teepublic');

    return (
      <div className={classes}>
        <CartButton
          cartUrl={this.props.configuration.cartUrl()}
          onCheckout={cartItems => console.log(cartItems)}
        />

        <StoreFilter
          albums={albums}
          productTypes={productTypes}
          onAlbumChange={this.albumChangeHandler}
          onProductTypeChange={this.productTypeChangeHandler}
          selectedAlbumId={selectedAlbumId}
          selectedProductTypeName={selectedProductTypeName}
        />

        <div className={`${CLASS_ROOT}__collection-wrap`}>
          <StoreFilter
            albums={albums}
            productTypes={productTypes}
            onAlbumChange={this.albumChangeHandler}
            onProductTypeChange={this.productTypeChangeHandler}
            selectedAlbumId={selectedAlbumId}
            selectedProductTypeName={selectedProductTypeName}
            filterStyle="nav"
          />

          <DesignCollection
            designs={designs}
            tileSize="large"
            buyProductLinkBuilder={this.props.configuration.buyProductUrl}
            storeId={this.props.storeData.id}
          />
        </div>

        <Pagination
          currentPage={selectedPage}
          totalPages={totalPages}
          onPageChange={this.pageChangeHandler}
        />
      </div>
    );
  }
}

Store.propTypes = {
  className: PropTypes.string,
  storeData: PropTypes.object.isRequired,
  configuration: PropTypes.shape({
    storeUrl: PropTypes.func.isRequired,
    cartUrl: PropTypes.func.isRequired,
    buyProductUrl: PropTypes.func.isRequired
  }).isRequired,
  selectedAlbumId: PropTypes.number,
  selectedPage: PropTypes.number,
  selectedProductTypeName: PropTypes.string
};
