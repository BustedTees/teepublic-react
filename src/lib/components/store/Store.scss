.tp-store__collection-wrap {
  display: flex;
  margin-top: 25px;
  align-items: flex-start;
}
