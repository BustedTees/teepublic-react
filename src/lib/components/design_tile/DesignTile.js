import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import Row from '../row/Row';
import Column from '../column/Column';
import MoneyHelper from '../../utils/MoneyHelper';
import _ from 'underscore';

import './DesignTile.css';

const CLASS_ROOT = 'tp-design-tile';

const DESIGN_IMAGE = {
  small: 200,
  medium: 300,
  large: 400
};

export default class DesignTile extends Component {
  render() {
    const { className, design, size, onClick } = this.props;

    const classes = classnames(
      CLASS_ROOT,
      {
        [`${CLASS_ROOT}--${size}`]: size
      },
      className
    );
    const owner = design._embedded.owner;
    const sku = design._embedded.defaultProduct._embedded.defaultSku;
    const { images, price } = sku;

    var skuMockupImage = _.find(images, function(image) {
      return image.type === 'mockup';
    });

    const designImage = (
      <img
        className={`${CLASS_ROOT}__image`}
        src={skuMockupImage.url}
        alt={skuMockupImage.type}
        height={DESIGN_IMAGE[size]}
      />
    );
    const designTitle = (
      <a className={`${CLASS_ROOT}__title`}>{design.title}</a>
    );
    const designPrice = (
      <p className={`${CLASS_ROOT}__price`}>
        {new MoneyHelper(price, 'USD').commaSeprated()}
      </p>
    );
    const ownerName = (
      <p className={`${CLASS_ROOT}__owner`}>{owner.username}</p>
    );
    const buyButton = <button>Buy</button>;

    return (
      <Column
        onClick={() => onClick()}
        className={classes}
        justify="start"
        align="center"
      >
        {designImage}
        <Row
          className={`${CLASS_ROOT}__details`}
          justify="between"
          align="start"
        >
          <Column
            className={`${CLASS_ROOT}__title-owner`}
            justify="start"
            align="start"
          >
            {designTitle}
            {ownerName}
            {designPrice}
          </Column>
          {buyButton}
        </Row>
      </Column>
    );
  }
}

DesignTile.propTypes = {
  size: PropTypes.oneOf(['small', 'medium', 'large']),
  design: PropTypes.object.isRequired,
  onClick: PropTypes.func.isRequired
};

DesignTile.defaultProps = {
  size: 'large'
};
