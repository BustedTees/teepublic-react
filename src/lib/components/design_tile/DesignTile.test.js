import React from 'react';
import ReactDOM from 'react-dom';
import DesignTile from './DesignTile';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import store from '../../../demo/StoreData';

describe('Design Tile Testing', () => {
  const mockDesign = store._embedded.designs[0];
  const mockProductType =
    mockDesign._embedded.defaultProduct._embedded.defaultSku.productType;
  const mockStoreId = store.id;
  const mockBuyProductLinkBuilder = jest.fn(
    (designId, productType, storeId) => {
      return `store/${storeId}/design/${designId}/product/${productType}`;
    }
  );
  const designTile = (
    <DesignTile
      design={mockDesign}
      buyProductLinkBuilder={mockBuyProductLinkBuilder}
      storeId={mockStoreId}
    />
  );

  it('DesignTile renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(designTile, div);
  });

  it('renders correctly', () => {
    const tree = renderer.create(designTile).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calls buyProductLinkBuilder once', () => {
    const tile = shallow(designTile);
    expect(mockBuyProductLinkBuilder).toBeCalledWith(
      mockDesign.id,
      mockProductType,
      mockStoreId
    );
  });

  it('Image click routes to BuyProduct link', () => {
    const tile = shallow(designTile);
    const imageLink = tile.find('.tp-design-tile__image-link').props().href;
    expect(imageLink).toBe(
      mockBuyProductLinkBuilder(mockDesign.id, mockProductType, mockStoreId)
    );
  });

  it('Buy Button routes to BuyProduct link', () => {
    // Hacky way to test it, could't get window location to work
    const tile = shallow(designTile);
    const buyOnClick = tile
      .find('button')
      .prop('onClick')
      .toString();
    expect(buyOnClick).toBe('() => window.location.href = buyProductLink');
  });
});
