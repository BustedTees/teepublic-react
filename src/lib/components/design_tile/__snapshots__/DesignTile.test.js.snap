// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Design Tile Testing renders correctly 1`] = `
<div
  className="tp-design-tile teepublic"
>
  <a
    className="tp-design-tile__image-link"
    href="store/146747/design/2310768/product/t-shirt"
  >
    <img
      alt="mockup"
      className="tp-design-tile__image"
      src="https://res.cloudinary.com/teepublic/image/private/s--VlT5WX6b--/t_Resized Artwork/c_crop,x_10,y_10/c_fit,w_649/c_crop,g_north_west,h_626,w_470,x_89,y_36/g_north_west,u_upload:v1462829015:production:blanks:mtl53ofohwq5goqjo9ke,x_-395,y_-325/b_rgb:eeeeee/c_limit,f_jpg,h_630,q_90,w_630/v1516963941/production/designs/2310768_0.jpg"
    />
  </a>
  <div
    className="tp-design-tile__details"
  >
    <p
      className="tp-design-tile__title"
    >
      <a
        href="store/146747/design/2310768/product/t-shirt"
      >
        Ring System
      </a>
    </p>
    <button
      className="tp-design-tile__buy-btn"
      onClick={[Function]}
    >
      Buy
    </button>
  </div>
  <p
    className="tp-design-tile__price"
  >
    $20.00
  </p>
</div>
`;
